src/Workspaces/SharedUtilitiesAndExtensions/Workspace/VisualBasic/LanguageServices/VisualBasicSyntaxGeneratorInternal.vb

```diff
@@ -126,6 +127,24 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.CodeGeneration
             Return If(typeContext, typeSymbol.GenerateTypeSyntax(), typeSymbol.GenerateExpressionSyntax())
         End Function
 
+        Public Overrides Function NegateEquality(generator As SyntaxGenerator, node As SyntaxNode, left As SyntaxNode, negatedKind As Operations.BinaryOperatorKind, right As SyntaxNode) As SyntaxNode
+            If negatedKind = BinaryOperatorKind.Equals Then
+                If node.IsKind(SyntaxKind.EqualsExpression, SyntaxKind.NotEqualsExpression) Then
+                    Return generator.ValueEqualsExpression(left, right)
+                Else
+                    Return generator.ReferenceEqualsExpression(left, right)
+                End If
+            ElseIf negatedKind = BinaryOperatorKind.NotEquals Then
+                If node.IsKind(SyntaxKind.EqualsExpression, SyntaxKind.NotEqualsExpression) Then
+                    Return generator.ValueNotEqualsExpression(left, right)
+                Else
+                    Return generator.ReferenceNotEqualsExpression(left, right)
+                End If
+            Else
+                Throw ExceptionUtilities.UnexpectedValue(negatedKind)
+            End If
+        End Function
```

Comment:
this is the main part of the fix.  In VB, when negating an equality expression, we defer to the original syntax to decide what to negate to.  so `=` becomes `<>` and `Is` becomes `IsNot` (and vice versa).